internal/session/sessionwantsender.go

```diff
@@ -298,16 +299,34 @@ func (sws *sessionWantSender) trackWant(c cid.Cid) {
 // processUpdates processes incoming blocks and HAVE / DONT_HAVEs.
 // It returns all DONT_HAVEs.
 func (sws *sessionWantSender) processUpdates(updates []update) []cid.Cid {
-	prunePeers := make(map[peer.ID]struct{})
-	dontHaves := cid.NewSet()
+	// Process received blocks keys
```

Comment:
Here I just moved things around so that once a block is received, we no longer try to process any HAVEs or DONT_HAVEs for the block